File: 1525. Number of Good Ways to Split a String.py
class Solution:
    def numSplits(self, s: str) -> int:
        left = right = ans = 0
        l_count, r_count = {}, {}
        
        for c in s:
            if c not in r_count:
                r_count[c] = 1
            
            else:
                r_count[c] += 1
        
        for c in s:
            if c not in l_count:
                l_count[c] = 1
            
            else:
                l_count[c] += 1
            
            r_count[c] -= 1
            
            if r_count[c]==0:
                del r_count[c]
            
            if len(l_count)==len(r_count):
                ans += 1
            
        return ans

File: 1525. Number of Good Ways to Split a String.js
/**
 * @param {string} s
 * @return {number}
 */
var numSplits = function(s) {
    let left = right = ans = 0;
    let [l_count, r_count] = [new Object(), new Object()];

    for(c of s){
        if(!Object.keys(r_count).includes(c)){
            r_count[c] = 1;
        }else{
            r_count[c] += 1;
        }
    }
    for(c of s){
        if(!Object.keys(l_count).includes(c)){
            l_count[c] = 1;
        }else{
            l_count[c] += 1;
        }
        r_count[c] -= 1;
        
        if(r_count[c]==0){
            delete r_count[c];
        }
        if(Object.keys(l_count).length==Object.keys(r_count).length){
            ans += 1;
        }
    }
    return ans;
};